core-tests/src/test/java/org/mule/runtime/core/api/util/IOUtilsTestCase.java

```diff
@@ -206,22 +201,31 @@ public void convertsToStringWithEncoding() throws Exception {
   }
 
   private URLConnection mockURLConnection(URL url) throws Exception {
-    PowerMockito.spy(IOUtils.class);
-    AtomicReference<URLConnection> connection = new AtomicReference<>();
-
-    url = spy(url);
-    when(IOUtils.class, "getResourceAsUrl", anyString(), any(Class.class), anyBoolean(), anyBoolean())
-        .thenReturn(url);
-
-    when(url.openConnection()).then(a -> {
-      URLConnection conn = spy((URLConnection) a.callRealMethod());
-      connection.set(conn);
-      return conn;
-    });
-
-    getResourceAsStream(RESOURCE_NAME, getClass());
-
-    return connection.get();
+    MockedStatic<IOUtils> utilities = Mockito.mockStatic(IOUtils.class);
+    try {
```

Comment:
use the try-with-resources mechanism, something like this (taken an example from the tutorial):

    try (MockedStatic<StaticUtils> utilities = Mockito.mockStatic(StaticUtils.class)) {
        utilities.when(StaticUtils::name).thenReturn("Eugen");
        assertThat(StaticUtils.name()).isEqualTo("Eugen");
    }

This does the close without explicitely doing it.